### PR TITLE
Add direct HTML input support

### LINKED ISSUE #3: Allow direct HTML input
[Issuehunt badges](https://issuehunt.io/r/sindresorhus/capture-website/issues/3)

It would be useful to be able to just pass a HTML string. The problem is that we can't really overload the `url` argument to accept a HTML string as it would make the parsing too ambiguous. Well, we *could* detect `<`, but then we would fail if the user thought they could just pass in a string like `Hello` without any tags. Better to make it explicit, I think.

Puppeteer docs: https://github.com/GoogleChrome/puppeteer/blob/v1.12.0/docs/api.md#pagesetcontenthtml-options

We should also add support for this in the [CLI tool](https://github.com/sindresorhus/capture-website-cli).

There are two alternatives I can think of:

#### 1. Method

`.html()` will return an object with a `html` property and a `Symbol` we check for internally so we can handle it correctly.

```js
await captureWebsite.file(captureWebsite.html('<h1>🦄</h1>'), 'screenshot.png');
```

#### 2. Option

```js
await captureWebsite.file('<h1>🦄</h1>', 'screenshot.png', {inputType: 'html'});
```
---

I'm happy to consider other solutions too.

---

> *Note: This issue has a bounty, so it's expected that you are an experienced programmer and that you give it your best effort if you intend to tackle this. Don't forget, if applicable, to add tests, docs (double-check for typos), and update TypeScript definitions. And don't be sloppy. Review your own diff multiple times and try to find ways to improve and simplify your code. Instead of asking too many questions, present solutions. The point of an issue bounty is to reduce my workload, not give me more. Include a 🦄 in your PR description to indicate that you've read this. Thanks for helping out 🙌* - @sindresorhus

---

<details>
<summary>
<b>IssueHunt Summary</b>
</summary>

### Backers (Total: $40.00)

- [issuehunt issuehunt](https://issuehunt.io/u/issuehunt) ($40.00)
### Submitted pull Requests
- [#33 Add direct HTML input support](https://issuehunt.io/r/sindresorhus/capture-website/pull/33)
---

### Tips

- Checkout the [Issuehunt explorer](https://issuehunt.io/r/sindresorhus/capture-website/) to discover more funded issues.
- Need some help from other developers? [Add your repositories](https://issuehunt.io/r/new) on IssueHunt to raise funds.
---
IssueHunt has been backed by the following sponsors. [Become a sponsor](https://issuehunt.io/membership/members)
</details>

### PR DESCRIPTION
**Pull request by fisker**
fixes #3

🦄

---

<details>
<summary>
<b>IssueHunt Summary</b>
</summary>

### Referenced issues

This pull request has been submitted to:
- [#3: Allow direct HTML input](https://issuehunt.io/repos/169625338/issues/3)
---

IssueHunt has been backed by the following sponsors. [Become a sponsor](https://issuehunt.io/membership/members)
</details>

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -28,6 +28,13 @@ declare namespace captureWebsite {
 	}
 
 	interface Options {
+		/**
+		Image type.
```
Comment (sindresorhus) [inline index.d.ts]: Incorrect

Can you also add an example code block of using the `html` option? (Readme too)

File: index.d.ts [hunk 2]
```diff
@@ -235,9 +242,9 @@ declare const captureWebsite: {
 	readonly devices: string[];
 
 	/**
-	Capture a screnshot of the given `url` and save it to the given `outputFilePath`.
+	Capture a screenshot of the given `input` and save it to the given `outputFilePath`.
 
-	@param url - The URL, file URL, data URL, or local file path to the website.
+	@param input - The URL, file URL, data URL, or local file path to the website, or HTML content of the website.
```
Comment (sindresorhus) [inline index.d.ts]:
```suggestion
	@param input - The URL, file URL, data URL, local file path to the website, or HTML.
```

**Comment by sindresorhus** [comment]
Can you also do a PR to add this to https://github.com/sindresorhus/capture-website-cli ?

**Comment by fisker** [comment]
> Can you also do a PR to add this to https://github.com/sindresorhus/capture-website-cli ?

Will do it later, I opened an issue there https://github.com/sindresorhus/capture-website-cli/issues/13

File: index.d.ts [hunk 3]
```diff
@@ -246,31 +253,35 @@ declare const captureWebsite: {
...
 		outputFilePath: string,
 		options?: captureWebsite.FileOptions
 	): Promise<void>;
 
 	/**
-	Capture a screnshot of the given `url`.
+	Capture a screenshot of the given `input`.
 
-	@param url - The URL, file URL, data URL, or local file path to the website.
+	@param input - The URL, file URL, data URL, or local file path to the website, or HTML content of the website.
```
Comment (sindresorhus) [inline index.d.ts]: You need to update this and the other sentence too. I thought that was implied when I commented on the first one.

Comment (fisker) [reply index.d.ts]: Double checked.

**Pull request merged**